frontend/src/components/RoomsList.jsx

```diff
@@ -3,22 +3,47 @@ import { Link as RouterLink } from 'react-router-dom';
 import PropTypes from 'prop-types';
 
 import {
-  List, ListItem, ListItemButton, ListItemText, Typography, IconButton,
+  Grid, Card, CardContent, CardActions, Typography, IconButton,
 } from '@mui/material';
 import {
   Delete as DeleteIcon,
   Edit as EditIcon,
 } from '@mui/icons-material';
 
 import { useDispatch } from 'react-redux';
+import styled from 'styled-components';
+import Button from '@mui/material/Button';
 import { deleteRoom } from '../actions';
 
+const StyledButton = styled(Button)`
+  &&.custom-button {
+    background-color: #652ead;
+    color: #ffffff;
+  }
+  &&.custom-button:hover {
+    background-color: #391052;
+  }
+  &&.custom-button:disabled {
+    background-color: #cccccc;
+  }
+`;
+
+const StyledCard = styled(Card)`
+  &&.custom-card {
+    border: 1px solid #f0e4ff;
```

Comment:
When styling MUI components, also consider having a look at [the `sx` prop](https://mui.com/material-ui/react-box/#the-sx-prop), which has access to the theme palette, among other things. This saves you from having to hunt down wherever you set custom colours, if you want to make changes to the theme later.